### DISCUSSION TITLE
lotus-miner storage attach --init --weight, It doesn't seem to work very well

### DISCUSSION THREAD

**Discussion by quantumgw** [filecoin-project/lotus#5602] [General Q&A] [answered]
How to prevent disks with utilization rate over 95% to continue to store sectors? The weight of new disks with utilization rate of 0% and old disks is 10. Also tried to increase the weight.But miner has been randomly storing sectors to these two locations. What should I do？

Can we add a new option? for example
/storage path/sectorstore.json
Max using storage available ratio = 95

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
https://filecoinproject.slack.com/archives/C0179RNEMU4/p1601378224411200?thread_ts=1601378070.410800&cid=C0179RNEMU4

> Storage is picked by multiplying % available space by the weight - so a 90% full disk with weight 10 is as likely to be picked as 0% full disk with weight 1 (10% * 10 vs 100% * 1)

**Comment by William8Work** [comment]
The weight setting doesn't work from the command line. I ran this:

`$ lotus-miner storage attach --init --store /media/data2/lotusstorage --weight 20`

The storage path is added and initialized. However, the storage.json shows weight = 10, the default value. Something wrong with the weight setting.

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment, marked as answer]
try put the `--weight ` option before the argument

`$ lotus-miner storage attach --init --store --weight 20 /media/data2/lotusstorage`

```
lotus-miner storage attach --help
NAME:
   lotus-miner storage attach - attach local storage path

USAGE:
   lotus-miner storage attach [command options] [arguments...]

[...cut...]
```

**Reply by William8Work** [reply]
ok. thank you :-)
